miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/MetadataApi.kt

```diff
@@ -0,0 +1,38 @@
+package com.rakuten.tech.mobile.miniapp.api
+
+import androidx.annotation.Keep
+import com.google.gson.annotations.SerializedName
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+internal interface MetadataApi {
+    @GET("host/{hostId}/miniapp/{miniappId}/version/{versionId}/{testPath}/metadata")
+    fun fetchMetadata(
+        @Path("hostId") hostId: String,
+        @Path("miniappId") miniAppId: String,
+        @Path("versionId") versionId: String,
+        @Path("testPath") testPath: String = ""
+    ): Call<MetadataEntity>
+}
+
+@Keep
+internal data class MetadataEntity(
+    @SerializedName("metadata") val metadata: MetadataResponse
```

Comment:
Seems that the Demo App is crashing when the `metadata` field isn't present. It's strange because it should instead throw an error which is caught by the Host App (a `MiniAppSdkException` type error)

Reply:
Yeah, looks like its a NPE has been caught because `MetadataResponse` is non-null here. I have changed it to nullable and handle the exception in `MiniAppDownloader.prepareMiniAppManifest`. Sorry, I didn't think about it before 😢